exchanges/kucoin/kucoin_futures_types.go

```diff
@@ -14,61 +14,86 @@ var validGranularity = []string{
+	SettlementSymbol                   string          `json:"settlementSymbol"`
+	Status                             string          `json:"status"`
+	FundingFeeRate                     float64         `json:"fundingFeeRate"`
+	PredictedFundingFeeRate            float64         `json:"predictedFundingFeeRate"`
+	DailyInterestRate                  float64         `json:"dailyInterestRate"`
+	FundingRateGranularity             int64           `json:"fundingRateGranularity"`
+	FundingRateCap                     float64         `json:"fundingRateCap"`
+	FundingRateFloor                   float64         `json:"fundingRateFloor"`
+	Period                             float64         `json:"period"`
+	EffectiveFundingRateCycleStartTime types.Time      `json:"effectiveFundingRateCycleStartTime"`
+	CurrentFundingRateGranularity      int64           `json:"currentFundingRateGranularity"`
+	OpenInterest                       types.Number    `json:"openInterest"`
+	TurnoverOf24h                      float64         `json:"turnoverOf24h"`
+	VolumeOf24h                        float64         `json:"volumeOf24h"`
+	MarkPrice                          float64         `json:"markPrice"`
+	IndexPrice                         float64         `json:"indexPrice"`
+	LastTradePrice                     float64         `json:"lastTradePrice"`
+	NextFundingRateTime                int64           `json:"nextFundingRateTime"` // Not a timestamp
+	NextFundingRateDateTime            types.Time      `json:"nextFundingRateDateTime"`
+	MaxLeverage                        float64         `json:"maxLeverage"`
+	SourceExchanges                    []string        `json:"sourceExchanges"`
+	PremiumsSymbol1M                   string          `json:"premiumsSymbol1M"`
+	PremiumsSymbol8H                   string          `json:"premiumsSymbol8H"`
+	FundingBaseSymbol1M                string          `json:"fundingBaseSymbol1M"`
+	FundingQuoteSymbol1M               string          `json:"fundingQuoteSymbol1M"`
+	LowPrice                           float64         `json:"lowPrice"`
+	HighPrice                          float64         `json:"highPrice"`
+	PriceChangePercentage              float64         `json:"priceChgPct"`
+	PriceChange                        float64         `json:"priceChg"`
+	K                                  float64         `json:"k"`
+	M                                  float64         `json:"m"`
+	F                                  float64         `json:"f"`
+	MMRLimit                           float64         `json:"mmrLimit"`
+	MMRLeverageConstant                float64         `json:"mmrLevConstant"`
+	SupportCross                       bool            `json:"supportCross"`
+	BuyLimit                           float64         `json:"buyLimit"`
+	SellLimit                          float64         `json:"sellLimit"`
+	AdjustK                            json.RawMessage `json:"adjustK"`              // Null value returned
```

Comment:
Since we can support null as zero values with times and numbers, I reckon we fill these out with types.Number and types.Time
<img width="994" height="418" alt="image" src="https://github.com/user-attachments/assets/f0b0b16e-2287-465f-a239-b7584e4b2cd5" />

Reply:
https://github.com/thrasher-corp/gocryptotrader/pull/2124/commits/55514b462416e84a9b4dc5af4d9292b08e27aeba